### DISCUSSION TITLE
Statistics

### DISCUSSION THREAD

**Discussion by fdw** [IgnisDa/ryot#302] [Ideas]
Do you plan to calculate show statistics like most watched genres or histograms over time (how much you watched in July, for example)? I think the data should already be available, but I haven't found it in the demo or an issue for it.

**Comment by IgnisDa** [comment]
Can you explain a bit more on what all features you'd want?

**Reply by fdw** [reply]
I haven't really thought about "all" features, but I can try to list some things for movies and series that come to mind. They vary in usefulness, though 😉

- What are my most watched genres?
- What are my highest rated genres?
- When do I watch movies/episodes? (Months/Hours/Weekdays/Years)
- Which actors/directors do I watch most?
- From which years/decades are my movies/series?
- What's the average rating of a director's/actor's movies? Which are the highest rated directors/actors by this metric?

**Reply by IgnisDa** [reply]
These sound nice. Do you have a UI in mind? Or examples of projects that do this well?

**Reply by fdw** [reply]
Trakt shows a graph for the genres of everything you watched in the last 30 days, that looks quite nice (in that context):
https://miro.medium.com/v2/resize:fit:4800/format:webp/1*1RUoLKi26hyEwWQLdwpUKQ.png

There's also a nice histogram.

**Reply by IgnisDa** [reply]
@fdw I have been looking into this. Can you send me a link on how I can access this for my own trakt account? Or do I need trakt premium for it?

**Comment by fdw** [comment]
The graph I've linked above should be on your Trakt dashboard. There are some more graphs on your Trakt profile page. But the more advanced/interesting ones need premium, unfortunately.

**Reply by IgnisDa** [reply]
Could you post screenshots of a few more graphs? I'll try to see if they'd be worth implementing for ryot.

**Reply by fdw** [reply]
Here are some samples for all time stats: https://blog.trakt.tv/all-time-year-in-review-f6f931e4461d . My favorites:
- The histogram of release years
- The histogram of my ratings
- The stacked bar of genres
- Plays by year

Some of my other original suggestions are actually my ideas and not from Trakt, so I can't show you an example 😉

**Reply by IgnisDa** [reply]
Would you be willing to share you trakt credentials with me? I would like to add these features to Ryot too but won't be able to work with just screenshots and descriptions. It's completely understandable if you can't :).

**Reply by fdw** [reply]
I'm sorry, but I won't share my credentials.
But what exactly are you interested in or how can I help you? I would think you'd need to query the database for the statistics, and then a way to show them nicely. Where does a Trakt account help?

**Reply by IgnisDa** [reply]
Yeah I need to see

- types of charts trakt has
- how does interaction with them work
- are they configurable? For eg, can I choose if the chart is over a month or overy a year

can't think of anything else but more will come up as i work on the feature

**Reply by fdw** [reply]
I'll try my best 🙂 But in general, I don't want to re-build what Trakt has, but really let ryot shine on its own.

# Types of charts
As I said, I'm mostly interested in:
- The histogram of release years
- The histogram of my ratings
- The stacked bar of genres
- Histogram of plays by year (play count and minutes watched)

# Interaction
I'd start with only a tooltip showing the exact number on hover. There is no further interaction possible.

# Configuration
On Trakt, I don't think you can choose the time range. However, that could be an interesting feature in the future.
Additionally, I'd like to look at movies and shows separately, so either two separate graphs or one where you can choose. (Advanced feature: stacked bars in the histogram.)

Hope that helps!
